src/main/scala/edu/tulane/cs/hetml/nlp/sprl/Triplets/TripletSentenceLevelConstraints.scala

```diff
@@ -36,6 +36,22 @@ object TripletSentenceLevelConstraints {
           (sentences(s) ~> sentenceToPhrase).toList._exists { x: Phrase => TrajectorRoleClassifier on x is "Trajector" }
         )
   }
+  val sim = new SegmentPhraseSimilarityClassifier()
+  val boostTrajectorByImage = ConstrainedClassifier.constraint[Sentence] {
+    var a: FirstOrderConstraint = null
+    s: Sentence =>
+      a = new FirstOrderConstant(true)
+      (sentences(s) ~> sentenceToPhrase).foreach {
+        p =>
+          val pairs = (phrases(p) ~> -segmentPhrasePairToPhrase).toList
+
+          a = a and
+            (
+              (TrajectorRoleClassifier on p is "Trajector") ==> pairs._exists(pair => sim on pair is "true")
+            )
+      }
+      a
+  }
```

Comment:
@Rahgooy do you want this PR to be merged still? it seems the constraints are not well-formulated.

Reply:
You mean it should be the other way around, right?

Reply:
@kordjamshidi, could you please clarify what you mean? thanks.